chasm/visibility.go

```diff
@@ -5,37 +5,27 @@ import (
 
 	commonpb "go.temporal.io/api/common/v1"
 	persistencespb "go.temporal.io/server/api/persistence/v1"
-	"go.temporal.io/server/common/payload"
 )
 
 const (
 	visibilityComponentFqType = "core.vis"
 	visibilityTaskFqType      = "core.visTask"
 )
 
-// CoreLibrary contains built-in components maintained as part of the CHASM framework.
-type CoreLibrary struct {
-	UnimplementedLibrary
+// VisibilitySearchAttributesProvider if implemented by the root Component,
+// allows the CHASM framework to automatically determine, at the end of
+// a transaction, if a visibility task needs to be generated to update the
+// visibility record with the returned search attributes.
+type VisibilitySearchAttributesProvider interface {
+	SearchAttributes(Context) map[string]VisibilityValue
```

Comment:
Mentioned this on slack too, but I'm not sure why we can't use the typed search attribute definitions from the SDK.

Reply:
Synced offline. Will keep this interface for now and for memo. Once SA registration is supported, this interface will be updated.